pkg/controller/persistence/dgraph/service.go

```diff
@@ -0,0 +1,37 @@
+package dgraph
+
+import (
+	"encoding/json"
+	"time"
+
+	api_v1 "k8s.io/api/core/v1"
+)
+
+type Service struct {
+	ID
+	IsService string    `json:"isService"`
+	Name      string    `json:"name,omitempty"`
+	StartTime time.Time `json:"startTime,omitempty"`
+	EndTime   time.Time `json:"endTime,omitempty"`
+	Pod       []*Pod    `json:"servicePods,omitempty"`
+}
+
+func PersistService(service api_v1.Service) error {
```

Comment:
Method instead of a function?

Reply:
Service object is not created yet, it's created in this method and persisted so can't write as method on Service.